src/tools/kata-ctl/src/exec.rs

```diff
@@ -0,0 +1,337 @@
+    hybrid_vsock_port: u64,
+) -> anyhow::Result<UnixStream> {
+    let stream = match UnixStream::connect(server_addr) {
+        Ok(s) => s,
+        Err(e) => {
+            return Err(anyhow!(e).context("failed to create named UNIX Domain stream socket"))
+        }
+    };
+
+    // when vport > 0, setup hvsock.
+    if hybrid_vsock_port > 0 {
+        setup_hybrid_vsock(stdin_handle, &stream, hybrid_vsock_port)?
+    }
+
+    Ok(stream)
+}
+
+fn debug_console_uds(sid: &str, sandbox_run_path: &str) -> PathBuf {
+    // we use pattern to get the full sandbox run path,
+    let sandbox_id = format!("{}{}", sid, "*");
+
+    // ${KATA_RUN_PATH}/root/{SANDBOX_ID}/kata.hvsock
+    // TODO: get vsock path for other scenario
+    let sandbox_path: String = [
+        sandbox_run_path,
+        sandbox_id.as_str(),
+        "root",
+        KATA_HYBRID_VSOCK,
+    ]
+    .join("/");
+
+    glob(sandbox_path.as_str())
+        .unwrap()
+        .filter_map(std::result::Result::ok)
+        .last()
+        .unwrap_or_else(|| PathBuf::from(sandbox_path.as_str()))
+}
+
+pub fn do_run_exec(sandbox_id: &str, debug_port: u64) -> anyhow::Result<()> {
+    let server_addr = debug_console_uds(sandbox_id, KATA_RUN_PATH)
+        .into_os_string()
+        .into_string()
+        .unwrap();
+
+    if !PathBuf::from(server_addr.clone()).exists() {
+        return Err(anyhow!("sandbox with run path {:?} not found", server_addr));
+    }
+
+    let mut epoll_context = EpollContext::new().expect("Fail to create epoll");
+    let stdin_handle = io::stdin();
+    epoll_context
+        .enable_stdin_event()
+        .expect("Fail to enable stdin");
+
+    let sock_addr: UnixStream = {
+        stdin_handle
+            .lock()
+            .set_raw_mode()
+            .expect("Fail to set stdin to RAW mode");
+
+        // by kata agent hvsock
+        create_ttrpc_socket(stdin_handle, server_addr, debug_port)?
+    };
+
+    epoll_context
+        .enable_debug_console_sock(sock_addr)
+        .expect("Fail to connect to the server");
+
+    epoll_context.process_handler().unwrap_or_default();
+    epoll_context.do_exit();
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::fs;
+    use std::path::PathBuf;
+
+    #[test]
+    fn test_debug_console_uds() {
+        let sandbox_id = "0a01abcd09xxxxx";
+        let sandbox_id2 = "0a02acec0axxxxx";
+        let sid = "0a01";
+
+        let sandbox01 = format!("{}/{}/root/kata.hvsock", "/tmp", sandbox_id);
+        let sandbox02 = format!("{}/{}/root/kata.hvsock", "/tmp", sandbox_id2);
+        // prepare clear env
+        fs::remove_dir_all(&sandbox01).unwrap_or_default();
+        fs::remove_dir_all(&sandbox02).unwrap_or_default();
+        // create dir
+        fs::create_dir_all(&sandbox01).unwrap();
+        fs::create_dir_all(&sandbox02).unwrap();
+
+        let expect_path = debug_console_uds(sid, "/tmp");
+        //let target_path = format!("{}/root/kata.hvsock", &sandbox01);
```

Comment:
Please remove this line, and actually I think test for sandbox02 is useless in current implementation

Reply:
OK, I will fix it.